src/mdns/mdns_mdnssd.cpp

```diff
@@ -1050,13 +1047,6 @@ void PublisherMDnsSd::ServiceSubscription::HandleGetAddrInfoResult(DNSServiceRef
     if (aErrorCode != kDNSServiceErr_NoError)
     {
         otbrLogWarning("DNSServiceGetAddrInfo failed: %d", aErrorCode);
-
-        mMDnsSd->OnServiceResolveFailed(*this, aErrorCode);
-    }
-    else if (mInstanceInfo.mAddresses.empty() && (aFlags & kDNSServiceFlagsMoreComing) == 0)
-    {
-        otbrLogDebug("DNSServiceGetAddrInfo reply: no IPv6 address found");
-        mInstanceInfo.mTtl = aTtl;
         mMDnsSd->OnServiceResolved(*this);
```

Comment:
Shouldn't this be `OnServiceResolveFailed`?

Reply:
Calling `OnServiceResolved` allows other instance info to be returned to Discovery Proxy, even though there is no IPv6 address.